core/serialization/deserializer/montage-interpreter.js

```diff
@@ -170,8 +172,10 @@ var MontageContext = Montage.specialize({
             }
 
             if (promises.length === 0) {
-                return Promise.resolve(this._invokeDidReviveObjects());
+                result = this._invokeDidReviveObjects();
```

Comment:
How is that working if the for() above happens to fill promises array while the interpreter has been set to be synchronous?

Reply:
Because `getObject` will throw an error instead of returning a promise if the context has been set to sync but some async work needs to be done.

Reply:
Because `getObject` will throw an error instead of returning a promise if the context has been set to sync but some async work needs to be done.